crowd_anki/config/config_settings.py

```diff
@@ -0,0 +1,74 @@
+from enum import Enum
+from dataclasses import dataclass
+
+from aqt import mw
+from ..utils.constants import USER_FILES_PATH
+
+@dataclass
+class ConfigSettings():
+    class Properties(Enum):
+        SNAPSHOT_PATH = "snapshot_path"
+        AUTOMATED_SNAPSHOT = "automated_snapshot"
+        SNAPSHOT_ROOT_DECKS = "snapshot_root_decks"
+        EXPORT_DECK_SORT_METHODS = "export_deck_sort_method"
+        EXPORT_DECK_SORT_REVERSED = "export_deck_sort_reversed"
+
+
+    class DeckExportSortMethods(Enum):
+        NO_SORTING = "none"
+        GUID = "guid"
+        FLAG = "flag"
+        TAG = "tag"
+        NOTEMODEL = "notemodel"
+        NOTEMODELID = "notemodelid"
+        FIELD1 = "field1"
+        FIELD2 = "field2"
+
+    snapshot_path: str
+    automated_snapshot: bool
+    snapshot_root_decks: list
+
+    export_deck_sort_reversed: bool
+    export_deck_sort_methods: list
+
+
+    def __init__(self, get_settings = True):
+        if get_settings:
+            self.config_file = mw.addonManager.getConfig(__name__)
+            self.get_all_values()
+
+    def get_all_values(self):
+        self.snapshot_path = self.config_file.get(self.Properties.SNAPSHOT_PATH.value, str(USER_FILES_PATH.resolve()))
+        self.automated_snapshot = self.config_file.get(self.Properties.AUTOMATED_SNAPSHOT.value, False)
+        self.snapshot_root_decks = self.config_file.get(self.Properties.SNAPSHOT_ROOT_DECKS.value, [])
+
+        self.export_deck_sort_reversed = self.config_file.get(self.Properties.EXPORT_DECK_SORT_REVERSED.value, False)
+        self.export_deck_sort_methods = self.config_file.get(self.Properties.EXPORT_DECK_SORT_METHODS.value, [self.DeckExportSortMethods.NO_SORTING.value])
+
+
+    def set_all_values(self):
+        self.config_file.update({
+            self.Properties.SNAPSHOT_PATH.value: self.snapshot_path,
+            self.Properties.AUTOMATED_SNAPSHOT.value: self.automated_snapshot,
+            self.Properties.EXPORT_DECK_SORT_REVERSED.value: self.export_deck_sort_reversed,
+
+            self.Properties.SNAPSHOT_ROOT_DECKS.value: self.snapshot_root_decks,
+            self.Properties.EXPORT_DECK_SORT_METHODS.value: self.export_deck_sort_methods
+        })
+
+    def write_values_to_anki(self):
+        mw.addonManager.writeConfig(__name__, self.config_file)
+    
+    def find_invalid_config_values(self):
+        self.handle_empty_textboxes()
+        
+        invalid_methods = [method for method in self.export_deck_sort_methods if method not in self.DeckExportSortMethods._value2member_map_]
+
+        return invalid_methods
+    
+    def handle_empty_textboxes(self):
+        if not self.export_deck_sort_methods[0]:
+            self.export_deck_sort_methods = [self.DeckExportSortMethods.NO_SORTING.value]
+        
+        if not self.snapshot_path:
+            self.snapshot_path = str(USER_FILES_PATH.resolve())
```

Comment:
how about making these properties instead and moving this logic to corresponding getters?

Reply:
hmm, not entirely sure it'd solve this, it'd provide the default values sure, but presumably you still want the check logic?

Reply:
I'll make your other change soon and we'll see.

Reply:
Fixed, as suggested. I just made a "set to default" function which takes a property. I only want it to be like this for these two text boxes, and I don't want it to be on all of them by default. Let me know and tell me what you think.

Reply:
's not quite what I meant. I had in mind having getters for this specific values that would encapsulate logic for retrieving their defaults. Something like:

```python
@attribute
def snapshot_path():
    return self._get(self.Properties.SNAPSHOT_PATH)

# so now if someone calls 

config.snapshot_path

# this function would be called
```

Reply:
Ah I see. Well, while trying to do this for each I encountered an issue: I need to set these values *locally* as a config_settings property, not in the proper config set. They only go in the config when they are being accepted and saved. For example when someone enters "guid, testindfsdkfjsd" as a sort method, this is saved as the local value of export_note_sort_methods in the config, so that it can perform verification on it on save. So since the UI files need to set them, I would have to also make a @property.setter on each, but they all are do damn simple apart from the export_note_sort_methods. 

So I settled on simply adding back in the original names properties, which will be set by load_values anyways, and adding a formatted_export_note_sort_methods property, which is called in NoteSorter now.

```python
class ConfigSettings:
    snapshot_path: str
    automated_snapshot: bool
    snapshot_root_decks: list
    export_notes_reverse_order: bool
    export_note_sort_methods: list

    @property
    def formatted_export_note_sort_methods(self) -> list:
        return [
            NoteSortingMethods(method)
            for method in self.export_note_sort_methods
        ]
```

Thoughts on this?

Reply:
Sure, some thoughts:

* You don't have to do it for all the fields, only for the ones that require custom verification logic/etc, and you can leave the other ones as is
* Also setters/getters can as well be automatically generated similar to how the fields are now. 
* The values are not written to file until you actually call `writeConfig` so I'm not sure that you actually need the intermediate storage